python/lsst/daf/butler/core/config.py

```diff
@@ -564,15 +563,14 @@ def _getKeyHierarchy(self, name):
             as a key in the Config it will be used regardless of the presence
             of any nominal delimiter.
         """
+        keys: list[str | int]
         if name in self._data:
-            keys = [
-                name,
-            ]
+            keys = [cast(str, name)]
```

Comment:
This is probably the right thing to do but in theory `name` could be a tuple -- the trick is that if `in` works directly there is no reason to try anything clever.

Reply:
Okay. Yes, `name` is anything but in this case we are looking in the internal dict and so yes it's only going to work if it's a string.